plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/index.tsx

```diff
@@ -249,9 +254,69 @@ const shouldBeUnmarkedAsConverted = ( props ) => {
 	);
 };
 
+const CollectionSpecificControls = (
+	props: ProductCollectionEditComponentProps
+) => {
+	const setQueryAttributeBind = useMemo(
+		() => setQueryAttribute.bind( null, props ),
+		[ props ]
+	);
+	const tracksLocation = useTracksLocation( props.context.templateSlug );
+	const trackInteraction = ( filter: FilterName ) => {
+		return recordEvent(
+			'blocks_product_collection_inspector_control_clicked',
+			{
+				collection: props.attributes.collection,
+				location: tracksLocation,
+				filter,
+			}
+		);
+	};
+	const queryControlProps = {
+		setQueryAttribute: setQueryAttributeBind,
+		trackInteraction,
+		query: props.attributes.query,
+	};
+
+	return (
+		<InspectorControls>
+			<PanelBody>
+				{
+					/**
+					 * Hand-Picked collection-specific controls.
+					 */
+					props.attributes.collection ===
+						'woocommerce/product-collection/hand-picked' && (
+						<HandPickedProductsControlField
+							{ ...queryControlProps }
+						/>
+					)
+				}
+			</PanelBody>
+		</InspectorControls>
+	);
+};
+
+const withCollectionSpecificControls =
+	< T extends EditorBlock< T > >( BlockEdit: ElementType ) =>
+	( props: ProductCollectionEditComponentProps ) => {
+		if ( ! isProductCollection( props.name ) || ! props.isSelected ) {
+			return <BlockEdit { ...props } />;
+		}
```

Comment:
Nit: I think we can also check for `props.attributes.collection` here instead of inside `CollectionSpecificControls` 🙂

Reply:
I intentionally moved the current collection check to `CollectionSpecificControls` in case other collections have their own controls in the future so that we're able to selectively render them as we do [here](https://github.com/woocommerce/woocommerce/blob/cd537befbeaee6a13849592c66d98f1019450721/plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/index.tsx#L165-L191). Does that make sense?